src/main/java/org/apache/sysml/api/mlcontext/MLResults.java

```diff
@@ -115,7 +120,21 @@ public Data getData(String outputName) {
 	 */
 	public MatrixObject getMatrixObject(String outputName) {
 		Data data = getData(outputName);
-		if (!(data instanceof MatrixObject)) {
+		if(data instanceof ScalarObject) {
```

Comment:
Okay, we may not need this anymore since the new `MLContext` can natively return scalar values.  Additionally, a simple call to `get("varName1", "varName2"...)` will return the correct types.  Checkout out this example where a scalar and a matrix from a script are returned in the correct formats:

```python
from SystemML import MLContext, dml

rdd1 = sc.parallelize(["1.0,2.0", "3.0,4.0"])

sums = """
s1 = sum(m1)
m2 = m1 * 2
"""

ml = MLContext(sc)
script = dml(sums).input(m1=rdd1).out("s1", "m2")
s1, m2 = ml.execute(script).get("s1", "m2")
s1, m2
```
Returns:
```python
(10.0, Matrix)
```

Then, 
```python
m2.toDF()
```
Returns
```python
DataFrame[ID: double, C1: double, C2: double]
````